editor/src/components/canvas/commands/adjust-css-length-command.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  BaseCommand,
+  CommandFunction,
+  CommandFunctionResult,
+  PathMappings,
+  TransientOrNot,
+} from './commands'
+
+export interface AdjustCssLengthProperty extends BaseCommand {
+  type: 'ADJUST_CSS_LENGTH_PROPERTY'
+  target: ElementPath
+  property: PropertyPath
+  valuePx: number
+  parentDimensionPx: number | undefined
+  createIfNonExistant: boolean
+}
+
+export function adjustCssLengthProperty(
+  transient: TransientOrNot,
+  target: ElementPath,
+  property: PropertyPath,
+  valuePx: number,
+  parentDimensionPx: number | undefined,
+  createIfNonExistant: boolean,
+): AdjustCssLengthProperty {
+  return {
+    type: 'ADJUST_CSS_LENGTH_PROPERTY',
+    transient: transient,
+    target: target,
+    property: property,
+    valuePx: valuePx,
+    parentDimensionPx: parentDimensionPx,
+    createIfNonExistant: createIfNonExistant,
+  }
+}
+
+export const runAdjustCssLengthProperty: CommandFunction<AdjustCssLengthProperty> = (
+  editorState: EditorState,
+  pathMappings: PathMappings,
+  command: AdjustCssLengthProperty,
+) => {
+  // Identify the current value, whatever that may be.
+  const currentValue: GetModifiableAttributeResult = withUnderlyingTargetFromEditorState(
+    command.target,
+    editorState,
+    left(`no target element was found at path ${EP.toString(command.target)}`),
+    (_, element) => {
+      return getModifiableJSXAttributeAtPath(element.props, command.property)
+    },
+  )
+  if (isLeft(currentValue)) {
+    return {
+      editorStatePatch: {},
+      pathMappings: pathMappings,
+      commandDescription: `Adjust Css Length Prop: ${EP.toUid(command.target)}/${PP.toString(
+        command.property,
+      )} not applied as value is not writeable.`,
+    }
+  }
+  const currentModifiableValue = currentValue.value
+  const simpleValueResult = jsxSimpleAttributeToValue(currentModifiableValue)
+  const valueProbablyExpression = isLeft(simpleValueResult)
+  const targetPropertyNonExistant: boolean = currentModifiableValue.type === 'ATTRIBUTE_NOT_FOUND'
+
+  if (targetPropertyNonExistant && !command.createIfNonExistant) {
+    return {
+      editorStatePatch: {},
+      pathMappings: pathMappings,
+      commandDescription: `Adjust Css Length Prop: ${EP.toUid(command.target)}/${PP.toString(
+        command.property,
+      )} not applied as the property does not exist.`,
+    }
+  }
+
+  if (valueProbablyExpression) {
+    // TODO add option to override expressions!!!
+    return {
+      editorStatePatch: {},
+      pathMappings: pathMappings,
+      commandDescription: `Adjust Css Length Prop: ${EP.toUid(command.target)}/${PP.toString(
+        command.property,
+      )} not applied as the property is an expression we did not want to override.`,
+    }
+  }
+
+  const parsePxResult = parseCSSPx(simpleValueResult.value) // TODO make type contain px
+  const parsePercentResult = parseCSSPercent(simpleValueResult.value) // TODO make type contain %
```

Comment:
You could put this one inside the else clause that checks `parsePxResult`, otherwise both always run.

Reply:
fixed!